clustermesh/clustermesh.go

```diff
@@ -1885,3 +1939,224 @@ func DisableWithHelm(ctx context.Context, k8sClient *k8s.Client, params Paramete
 	_, err = helm.Upgrade(ctx, k8sClient.RESTClientGetter, upgradeParams)
 	return err
 }
+
+func getRelease(kc *k8s.Client, namespace string) (*release.Release, error) {
+	client := kc.RESTClientGetter
+	release, err := helm.GetCurrentRelease(client, namespace, defaults.HelmReleaseName)
+	if err != nil {
+		return nil, err
+	}
+	return release, nil
+}
+
+// validateCAMatch determines if the certificate authority certificate being
+// used by aiLocal and aiRemote uses a matching keypair. The bool return value
+// is one of: nil (unknown), true (keypairs match), false (keys do not match)
+func (k *K8sClusterMesh) validateCAMatch(aiLocal, aiRemote *accessInformation) (*bool, error) {
+	caLocal := aiLocal.CA
+	block, _ := pem.Decode(caLocal)
+	if block == nil {
+		panic("failed to parse certificate PEM for local CA cert")
```

Comment:
Nit. It is likely better to return an error rather than panic here. Same below.

Reply:
Ah, of course, I meant to replace these